be/src/storage/inverted/clucene/clucene_inverted_writer.cpp

```diff
@@ -0,0 +1,327 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "clucene_inverted_writer.h"
+
+#include <CLucene/analysis/LanguageBasedAnalyzer.h>
+#include <CLucene/util/Misc.h>
+#include <fmt/format.h>
+
+#include <boost/locale/encoding_utf.hpp>
+
+#include "common/status.h"
+#include "storage/inverted/index_descriptor.hpp"
+#include "types/logical_type.h"
+#include "util/faststring.h"
+
+namespace starrocks {
+
+#define FINALIZE_OUTPUT(x) \
+    if (x != nullptr) {    \
+        x->close();        \
+        _CLDELETE(x);      \
+    }
+#define FINALLY_FINALIZE_OUTPUT(x) \
+    try {                          \
+        FINALIZE_OUTPUT(x)         \
+    } catch (...) {                \
+    }
+
+template <LogicalType field_type>
+class CLuceneInvertedWriterImpl : public CLuceneInvertedWriter {
+public:
+    using CppType = typename CppTypeTraits<field_type>::CppType;
+
+    explicit CLuceneInvertedWriterImpl(const std::string& field_name, const std::string& directory,
+                                       const TabletIndex* inverted_index)
+            : _directory(std::move(directory)), _inverted_index(inverted_index) {
+        _parser_type = get_inverted_index_parser_type_from_string(
+                get_parser_string_from_properties(_inverted_index->index_properties()));
```

Comment:
I tested the code, it seems that if I specify the parser in Upper case in SQL like: ("PARSER" = "standard"), I still get none. It seems that _inverted_index->index_properties() is case sensitive.